allensdk/brain_observatory/behavior/data_objects/base/_data_object_abc.py

```diff
@@ -0,0 +1,106 @@
+import abc
+from collections import deque
+from typing import Any, Optional, Set
+
+from pynwb import NWBFile
+
+from allensdk.brain_observatory.comparison_utils import compare_fields
+
+
+class DataObject(abc.ABC):
+    """An abstract class that prototypes properties that represent a
+    category of experimental data/metadata (e.g. running speed,
+    rewards, licks, etc.) and that prototypes methods to allow conversion of
+    the experimental data/metadata to and from various
+    data sources and sinks (e.g. LIMS, JSON, NWB).
+    """
+
+    def __init__(self, name: str, value: Any,
+                 exclude_from_equals: Optional[Set] = None):
+        """
+        :param name
+            Name
+        :param value
+            Value
+        :param exclude_from_equals
+            Optional set which will exclude these properties from comparison
+            checks to another DataObject
+        """
+        self._name = name
+        self._value = value
+
+        efe = exclude_from_equals if exclude_from_equals else set()
+        self._exclude_from_equals = efe
+
+    @property
+    def name(self) -> str:
+        return self._name
+
+    @property
+    def value(self) -> Any:
+        return self._value
+
+    @classmethod
+    def _get_vars(cls):
+        return vars(cls)
+
+    def to_dict(self):
```

Comment:
can you add a doctring with `Note` about what this while loop is doing?

Reply:
Sure, I am planning on adding tests for this method and adding docstring